### PR TITLE
Restructured the docs website to accommodate both user and dev docs

### PR DESCRIPTION
**Pull request by Pickysaurus**
Added some very basic user-facing docs.

### CODE REVIEW

File: docs/users/SystemRequirements.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+## Windows 
+
+|| Minimum | Recommended |
+| --- | --- | --- |
+| OS | Windows 10 or higher (64-bit OS required) | Windows 10/11 with updates |
+| Processor | Intel®️ Core™️ i5-7500 | Intel®️ Core™️ i7-7500 |
+| Memory | 4 GB RAM | 16 GB RAM |
+| Graphics | 2GB VRAM | 4GB VRAM |
+| Storage | 10GB available space* | 200GB available space* |
```
Comment (erri120) [inline docs/users/SystemRequirements.md:14]: Where did these values come from?

Comment (erri120) [reply docs/users/SystemRequirements.md:14]: These "requirements" don't really make a lot of sense. The only thing we require on Windows is Windows 10 or newer. We don't care what hardware the PC has.

File: docs/users/SystemRequirements.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+<sub>\* Storage space varies by the number of games managed, the size of the managed games and the mods installed.</sub>
+
+## SteamOS + Linux 
+|| Minimum | Recommended |
+| --- | --- | --- |
+| OS | Ubuntu 12.04 LTS | A currently supported Linux distribution |
+| Processor | Intel®️ Core™️ i5-7500 | Intel®️ Core™️ i7-7500 |
+| Memory | 4 GB RAM | 16 GB RAM |
+| Graphics | 2GB VRAM | 4GB VRAM |
+| Storage | 10GB available space* | 200GB available space* |
```
Comment (erri120) [inline docs/users/SystemRequirements.md:25]: This is also just straight up wrong. Ubuntu 12.04 is ancient, and **not** supported. I don't know where you got these values from, but you should just use the original requirements I wrote down.

Comment (Pickysaurus) [reply docs/users/SystemRequirements.md:25]: I can update this page to just have vague recommended specs but the app will not run well on a potato so it's important to set expectations.

Comment (erri120) [reply docs/users/SystemRequirements.md:25]: Just don't put any specs in there. Instead, you can write down that the App is bottlenecked by disk speed, so an SSD or NVME is recommended. The App is also multithreaded, and will make use of all available cores.

File: docs/styles.css [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/* Extra styles for the Nexus Mods App docs */
+
+.md-tabs {
+    background-color: var(--nexus-background-tertiary);
+}
+
+[dir=ltr] .md-header__title {
+    margin-left: 0
+}
```
Comment (erri120) [inline docs/styles.css:9]: This should probably be added to https://github.com/Nexus-Mods/NexusMods.MkDocsMaterial.Themes.Next.

File: docs/users/Uninstall.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+## Manual Uninstall (Fallback)
+
+!!! warning "Caution"
+    Manual uninstall should only be used as a fallback if the normal installer is not available. You will need to manually remove any mod files from your game folder(s).
+
+If something goes wrong during uninstall, it may be necessary to remove the app manually. The following steps must be performed manually by the user:
+
+1. Inside the App, ensure that all mods for your loadouts in all games are disabled and these changes have been applied. This will remove any files added to the game folder and get the game back to a "vanilla" state. (When possible, you should delete your loadouts).</li>
+2. Close the App. 
+3. Delete the contents of `%LOCALAPPDATA%\NexusMods.App` on Windows or `$XDG_STATE_HOME/NexusMods.App/Logs` on Linux to remove all downloaded mods and internal databases used by the app.
```
Comment (erri120) [inline docs/users/Uninstall.md:43]: On Linux, it's `$XDG_STATE_HOME/NexusMods.App` and `$XDG_DATA_HOME/NexusMods.App`.

**Review by erri120 — CHANGES_REQUESTED** [review]
See comments.

File: docs/developers/changelog-assets/README.md
Comment (erri120) [inline docs/developers/changelog-assets/README.md:1]: The `changelog-assets` folder should be inside `docs`, not `docs/developers`.

Comment (Pickysaurus) [reply docs/developers/changelog-assets/README.md:1]: This was not documented anywhere but is easy to fix.

File: docs/styles.css (hunk 3 above)
Comment (Pickysaurus) [reply docs/styles.css:9]: You're welcome to do so.

File: docs/developers/index.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+#### System Package Manager
+
+If you're want to create a package for a platform, see [Contributing](./Contributing.md#for-package-maintainers) for more details.
+
+[![Packaging status](https://repology.org/badge/vertical-allrepos/nexusmods-app.svg)](https://repology.org/project/nexusmods-app/versions)
+
+### macOS
+
+We currently don't publish releases for macOS.
```
[the remark this answers is not in the document]
Comment (Pickysaurus) [reply docs/developers/index.md:50]: I'll update it later. It's way to complex and developer-y for general users.

File: docs/developers/changelog-assets/README.md
Comment (erri120) [reply docs/developers/changelog-assets/README.md:1]: The directory is referenced by the changelog and by scripts directly, which is why checks are failing on the PR.

File: docs/developers/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (erri120) [reply docs/developers/index.md:50]: What about this is developer-y? These are requirements for running the App.

**Review by erri120 — CHANGES_REQUESTED** [review]
See open comments.

File: docs/users/SystemRequirements.md (hunk 1 above)
Comment (halgari) [reply docs/users/SystemRequirements.md]: We should either not have the requirements or track down actual values. I'm more in favor of the latter. I think we can remove all of the windows requirements except for Windows 10/11 for now, as that's about all we currently know.

File: docs/users/SystemRequirements.md (hunk 2 above)
Comment (halgari) [reply docs/users/SystemRequirements.md]: I have to agree here, saying we support Ubuntu 12 means we actually support it, and I seriously doubt we'll do that. I'm also not sure if we want to tie the minimum requirements to Ubuntu when Steam Deck runs arch. So we may just want to say "we support Linux" or something of that nature

File: docs/styles.css (hunk 3 above)
Comment (halgari) [reply docs/styles.css:9]: Well no, we want to keep the styling consistent across the projects.

File: docs/developers/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (halgari) [reply docs/developers/index.md:50]: I'm fine with these staying here for now, we can move it later easily enough

File: docs/styles.css (hunk 3 above)
Comment (halgari) [reply docs/styles.css:9]: Eh, scratch that, the perms on that repo are borked and need to be updated. I'll work on that sometime in the future

File: docs/users/SystemRequirements.md (hunk 2 above)
Comment (halgari) [reply docs/users/SystemRequirements.md]: fixed

File: docs/users/SystemRequirements.md (hunk 1 above)
Comment (halgari) [reply docs/users/SystemRequirements.md]: fixed

**Review by halgari — APPROVED** [review]

**Pull request merged**